packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/search/SearchByAssetTypeController.java

```diff
@@ -114,32 +109,12 @@ public ResponseEntity<List<JsonNode>> searchByAssetType(
 			KnnQuery knn = null;
 			if (text != null && !text.isEmpty()) {
 
-				// create the embedding search request
-				final GoLLMSearchRequest embeddingRequest = new GoLLMSearchRequest();
-				embeddingRequest.setText(text);
-				embeddingRequest.setEmbeddingModel(EMBEDDING_MODEL);
-
-				final TaskRequest req = new TaskRequest();
-				req.setTimeoutMinutes(REQUEST_TIMEOUT_MINUTES);
-				req.setType(TaskType.GOLLM);
-				req.setInput(embeddingRequest);
-				req.setScript("gollm_task:embedding");
-				req.setUserId(currentUserService.get().getId());
-
-				final TaskResponse resp = taskService.runTaskSync(req);
-
-				if (resp.getStatus() != TaskStatus.SUCCESS) {
-					throw new ResponseStatusException(
-							org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR,
-							"Unable to generate vectors for knn search");
-				}
-
-				final byte[] outputBytes = resp.getOutput();
-				final JsonNode output = objectMapper.readTree(outputBytes);
+				final TerariumAssetEmbeddings embeddings = embeddingService.generateEmbeddings(text);
 
-				final EmbeddingsResponse embeddingResp = objectMapper.convertValue(output, EmbeddingsResponse.class);
-
-				final List<Float> vector = embeddingResp.getResponse();
+				final List<Float> vector = Arrays.stream(
+								embeddings.getEmbeddings().get(0).getVector())
+						.mapToObj(d -> (float) d)
```

Comment:
Is this intentional? Elsewhere embeddings are declared as `double`s

Reply:
Yeah, the task returns doubles, but the java ES API wants floats...